File: src/Workflow.Core/Domain/K2/Worklist.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Workflow.DataContract.K2
{
    public class WorklistHeader
    {
        public int RequestHeaderId { get; set; }
        public int ProcInstId { get; set; }
        public string RequestCode { get; set; }
        public string Requestor { get; set; }
        public string ProcessName { get; set; }
        public DateTime LastActionDate { get; set; }
    }

    public partial class Worklist : WorklistHeader
    {
        private string _openFormUrl;
        public Worklist()
        {
            Actions = new List<string>();
        }

        public Worklist Update(WorklistHeader header)
        {
            RequestHeaderId = header.RequestHeaderId;
            RequestCode = header.RequestCode;
            Requestor = header.Requestor;
            ProcessName = header.ProcessName;
            LastActionDate = header.LastActionDate;

            return this;
        }

        public int ActInstDestId { get; set; }
        public string Serial { get; set; }
        public string Folio { get; set; }
        public string WorkflowPath { get; set; }
        public string WorkflowUrl {
            get
            {
                return string.Format(@"#k2/flow/{0}/{1}", ProcInstId, Folio);
            }
        }
        public string OpenFormUrl {
            get
            {
                if (!string.IsNullOrEmpty(_openFormUrl))
                {
                    var segments = _openFormUrl.Split('#');
                    if (segments.Length > 1) {
                        _openFormUrl = ("#" + segments[1]);
                    }
                }

                return _openFormUrl;
            }
            set
            {
                _openFormUrl = value;
            }
        }
        public string ActivityName { get; set; }
        public string Originator { get; set; }
        /* Owner user or current user */
        public string ManagedUser { get; set; }
        public string OpenedBy { get; set; }
        public string AllocatedUser { get;  set; }
        public DateTime? OpenedDateTime { get; set; }
        public string Status { get; set; }
        public int Priority { get; set; }
        public List<string> Actions { get; set; }
        [NotMapped]
        public bool IsOutOfficeInst {
            get {
                // Define as OOF for status that "Available", but allocated user is not current user
                return (ManagedUser.ToUpper() != AllocatedUser.ToUpper()) && (Status == "Available");
            }
        }
        [NotMapped]
        public bool IsEscalatable
        {
            get
            {
                // Set disabled escaltion on OOF process instance.
                // Set disabled escaltion on OPENED process instance.
                return !IsOutOfficeInst && (Status != "Open");
            }
        }
    }
}

File: src/Workflow.Web.Service/Workflow/app/model/common/worklists/Worklist.js
Ext.define('Workflow.model.common.worklists.Worklist', {
    extend: 'Ext.data.Model',
    local:true,
    fields: [
		{ name: 'requestorId',  type: 'int', mapping: 'requestor' },
		{ name: 'requestor',    type: 'string', mapping: 'requestor' },
		{ name: 'serial', type: 'string', mapping: 'serial' },
		{ name: 'originator', type: 'string', mapping: 'originator' },
		{ name: 'allocatedUser', type: 'string', mapping: 'allocatedUser' }, // TODO: fixed mising field
		{ name: 'activityName', type: 'string', mapping: 'activityName' },
		{ name: 'folio', type: 'string', mapping: 'folio' },
        { name: 'workflowName', type: 'string', mapping: 'workflowPath' },
        { name: 'processName' },
        { name: 'priority', type: 'int', mapping: 'priority' },
		{ name: 'activityStartDate', mapping: 'lastActionDate' },
		{ name: 'startDate', mapping: 'lastActionDate' },  // TODO: fixed mising field
		{ name: 'status', type: 'string', mapping: 'status' },
		{ name: 'viewFlow', type: 'string', mapping: 'workflowUrl' },
        { name: 'data', type: 'string', mapping: 'openFormUrl' },
		{ name: 'openedBy', type: 'string', mapping: 'openedBy' },
        { name: 'viewFormUrl', type: 'string', mapping: 'formPath',
            convert: function (value, record) {
                return Ext.String.format('#{0}/SN={1}_99999',value, record.get('procInstId'));
            }
        },
        //{ name: 'noneK2Form', type: 'boolean', mapping: 'NoneK2Form' },
        { name: 'sharedUser', type: 'string', mapping: 'sharedUser' },
        { name: 'managedUser', type: 'string', mapping: 'managedUser' },
        { name: 'isShared', type: 'boolean', mapping: 'delegated' }
    ],
	 proxy:{
        type:'rest',
        url: Workflow.global.Config.baseUrl + 'api/worklists',
        reader:{
            type: 'json',
            rootProperty: 'records'
        }
     } 
});
//requestHeaderId: 29500,
//procInstId: 6158,
//actInstDestId: 44,
//serial: "6158_44",
//folio: "ATCF-000030",
//requestCode: "ATCF_REQ",
//workflowName: "Additional Time Worked Claim Form",
//formPath: "atcf-request-form",
//activityName: "Requestor Rework",
//originator: "K2:NAGAWORLD\yimsamaune",
//requestor: "011541-YIM SAMAUNE ",
//openedBy: "K2:NAGAWORLD\YIMSAMAUNE",
//lastActionDate: "2017-11-16T11:26:19.93",
//status: "Allocated",
//priority: "Medium",
//actions: "Cancelled"

/*
 convert: function (value, record) {
    if (!Ext.isArray(value)) {
        return [];
    }
    else {
        return value;
    }
},
 */